### PR TITLE
Discontinuing the use of Weaviate modules

### PR DESCRIPTION
**Pull request by moekidev**
ref: https://github.com/andreibondarev/weaviate-ruby/pull/18#issuecomment-1554318857

I think we should embed via Langchain.rb and then passing it to Weaviate when integrating Langchain.rb and Weaviate.

https://weaviate.io/developers/weaviate/api/rest/objects#with-a-custom-vector

That way, we can align with other vector databases and APIs.
Furthermore, for Langchain users, the benefits of the Weaviate module are not that substantial.

### CODE REVIEW

**Comment by andreibondarev** [comment]
@moekidev Do you think this produces better results? By not relying on the Weaviate internal modules.

**Comment by moekidev** [comment]
For maintainability. It helps reduce [accidental reuse of LLM names](https://github.com/andreibondarev/langchainrb/blob/5796dc0c9bfcb2514f6ae2d6f5b3cea33bd486b3/lib/vectorsearch/weaviate.rb#L46) and [branching based on the presence or absence of modules](https://github.com/andreibondarev/langchainrb/blob/5796dc0c9bfcb2514f6ae2d6f5b3cea33bd486b3/lib/vectorsearch/weaviate.rb#L97-L117). Of course, it is possible to rely on the extensibility of Weaviate modules. However, currently, Langchain.rb is limited to OpenAI, Cohere, and HuggingFace APIs to align with other vector databases. As a result, it does not benefit from the full extent of extensibility. Nevertheless, I believe this is the right choice. It is advisable to have Langchain.rb designed in a way that allows for easy replacement of the database and language model, ensuring compatibility among APIs.

**Comment by andreibondarev** [comment]
@moekidev Could you please run `standardrb --fix` and commit to fix the failing specs?

File: lib/vectorsearch/weaviate.rb [hunk 1]
```diff
@@ -63,14 +58,9 @@ def create_default_schema
     # @param k [Integer|String] The number of results to return
     # @return [Hash] The search results
     def similarity_search(query:, k: 4)
-      near_text = "{ concepts: [\"#{query}\"] }"
+      embedding = generate_embedding(text: query)
```
Comment (andreibondarev) [inline lib/vectorsearch/weaviate.rb:61]: Right now we just use `llm_client.embed`. `generate_embedding` alias was removed.

File: spec/vectorsearch/weaviate_spec.rb [hunk 2]
```diff
@@ -55,11 +70,25 @@
...
+          {
+            "embedding" => [
+              -0.0018150936,
+              0.0017554426,
+              -0.022715086
+            ]
+          }
+        ]
+      })
     end
```
Comment (andreibondarev) [inline spec/vectorsearch/weaviate_spec.rb:93]: Also I would really appreciate if you can write the spec for the `#ask` method. Thank you!

**Comment by andreibondarev** [comment]
@moekidev A few comments and it's good to go! 🚀

File: spec/vectorsearch/weaviate_spec.rb (hunk 2 above)
Comment (moekidev) [reply spec/vectorsearch/weaviate_spec.rb:93]: OK. I wrote!

**Comment by moekidev** [comment]
@andreibondarev Please check.

**Comment by moekidev** [comment]
@andreibondarev Could you please review it again?

**Pull request merged**